processor/src/advice/providers.rs

```diff
@@ -184,6 +190,15 @@ where
             .map_err(ExecutionError::MerkleStoreMergeFailed)
     }
 
+    // TODO: Do we want to return a generic MerkleStore instead of a SimpleMerkleMap?
+    fn get_store_subset<I, R>(&self, roots: I) -> MerkleStore
```

Comment:
I haven't thought about it much. Seems like returning `MerkleStore` (the way it is now) is a bit clearer - but I don't really have a strong preference.

Reply:
Lets leave it as is as theres no obvious reason to make it generic at this point.